### ISSUE TITLE
ARM Deployment Validation Error

### ISSUE THREAD

**Issue by sedouard** [Azure/azure-sdk-for-node#1431]
I'm finding that any of the templates I try to deploy using the xplat-cli v0.9.3 with this command:

```
azure group deployment create sedouard-fleet-bootstrapper --template-file azuredeploy.json --parameters-file azuredeploy-parameters.json
```
(the resource group `sedouard-fleet-bootstrapper` does exist, btw)

Doesn't seem to work and fails with this error:

```
error:   The request content was invalid and could not be deserialized: 'Required property 'properties' not found in JSON. Path '', line 1, position 2.'.
info:    Error information has been recorded to /Users/steve/.azure/azure.err
error:   group deployment create command failed
```

What seems to be happening is that in [`resourceManagementClient.js:2348`](https://github.com/Azure/azure-sdk-for-node/blob/9c81cbd0de8907ef37d067af6e2da91a00ecdcee/lib/services/resourceManagement/lib/resourceManagementClient.js#L2348) the template validation code here is set for `api-version=2014-04-01-preview`. The validation code doesn't find a top-level field in the template called `properties` and ends up sending a POST to the validation api with an empty request body. Hence causing the error message above.

This started happening only after I've updated to the latest version. I'm not sure if this issue belongs in the x-plat CLI or if it should be something directed to the SDK?

**Comment by sedouard** [comment]
Seems as though it has something to do with this commit https://github.com/Azure/azure-sdk-for-node/commit/63cad66ca6554e20cc12679cefca5874dd7f9d65. The azure-cli v0.8.8 doesn't hit this issue  and uses the same code path.

**Comment by paulmey** [comment]
I hit the same issue (with @gabrielhartmann, @johngossman). It looks like https://github.com/Azure/azure-sdk-for-node/commit/63cad66ca6554e20cc12679cefca5874dd7f9d65 changed the expected format of the parameters object (parameters.template -> parameters.properties.template etc.).

**Comment by vivsriaus** [comment]
@sedouard @paulmey We have a fix out for this already. PR: https://github.com/Azure/azure-xplat-cli/pull/1820

Thanks for reporting the issue. I'll let you know when the fix is out. In the meantime, please use Azure CLI version 0.9.2 to get around this issue.

**Comment by sedouard** [comment]
Honestly, rolling back to 0.9.2 didn't work for me either. keep hitting this:

```
{ [Error: Cannot find module './lbShowUtil'] code: 'MODULE_NOT_FOUND' }
Error: Cannot find module './lbShowUtil'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/usr/local/lib/node_modules/azure-cli/lib/commands/arm/network/loadBalancer._js:5:18)
    at Module._compile (module.js:460:26)
    at Module.pModule._compile (/usr/local/lib/node_modules/azure-cli/node_modules/streamline/lib/compiler/register.js:72:15)
    at Object.streamliners._js [as ._js] (/usr/local/lib/node_modules/azure-cli/node_modules/streamline/lib/compiler/underscored.js:120:12)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
```

At least on my machine I can't get anywhere unless I rollback all the way to 0.8.8

**Comment by gabrielhartmann** [comment]
npm -g uninstall azure-cli
npm -g install azure-cli@0.9.2

The above worked for me.

**Comment by sedouard** [comment]
You probably aren't using a load balancer? @gabrielhartmann

**Comment by gabrielhartmann** [comment]
Yes, I haven't tried any deployments with a load balancer.

**Comment by sedouard** [comment]
I take it back version 0.8.15 published February 18th is the last version that works with a load balancer. This is definitely a regression as well. I'll file another bug against the CLI looks like it's in there.

**Comment by amarzavery** [comment]
@sedouard - Are you still seeing this issue? We published a new version of cli (0.9.15) yesterday. Please take a look and let us know.

**Comment by stackArmor** [comment]
I'm still seeing this issue when using the azure ask for node. It fails trying to validate a template with the error message property properties is missing. Please help.
